src/py/flwr/common/parametersrecord.py

```diff
@@ -50,17 +52,26 @@ class Array:
     data: bytes
 
 
+def _check_key(key: str) -> None:
+    """Check if key is of expected type."""
+    if not isinstance(key, str):
+        raise TypeError(f"Key must be of type str. You passed {type(key)}.")
+
+
+def _check_values(value: Array) -> None:
```

Comment:
Small suggestion since this function checks a single value:
```suggestion
def _check_value(value: Array) -> None:
```